extensions/functions_geometry.yaml

```diff
@@ -0,0 +1,35 @@
+%YAML 1.2
+---
+types:
+  - name: geometry
+    structure: "BINARY"
+#    description: |
+#      An opaque type that can represent one or many points, lines, or shapes encompassing
+#      2, 3 or 4 dimension.
+scalar_functions:
+  -
+    name: "point"
+    description: >
+      Returns a 2D point with the given `x` and `y` coordinate values.
+    impls:
+      - args:
+          - name: x
+            value: fp64
+          - name: y
+            value: fp64
+        return: u!geometry
+  -
+    name: "makeline"
+    description: >
+      Returns a linestring connecting the endpoint of geometry `x` to the begin point of
+      geometry `y`. Repeated points at the beginning of input geometries are collapsed to a single point.
+
+      A linestring can be closed or simple.  A closed linestring starts and ends on the same
+      point. A simple linestring does not cross or touch itself.
+    impls:
+      - args:
+          - name: x
+            value: u!geometry
+          - name: y
+            value: u!geometry
```

Comment:
Small nitpick (and I also don't know if substrait has something like positional args where the actual name doesn't matter, in which case it is only for documentation): you might want to use different argument names than `x` and `y`, since naively that would easily be interpreted as x and y coordinates (like in `point()`), while here it's rather "point 1" and "point 2". 
So I would rather go with something like `geom1`/`geom2`, or if preferring something generic, eg `a`/`b`

Reply:
Ahh...I think I missed this when we decided to change to using the opaque geometry type.  Thanks for catching this!  I'm addressing it in the PR: https://github.com/substrait-io/substrait/pull/554/files